shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -0,0 +1,167 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.h"
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#import "flutter/fml/logging.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController_Internal.h"
+
+// Method Channel name to start spell check.
+static NSString* const kInitiateSpellCheck = @"SpellCheck.initiateSpellCheck";
+
+@interface FlutterSpellCheckPlugin ()
+
+@property(nonatomic, assign) FlutterMethodChannel* methodChannel;
+@property(nonatomic, retain) UITextChecker* textChecker;
+
+@end
+
+@implementation FlutterSpellCheckPlugin
+
+- (instancetype)initWithMethodChannel:(FlutterMethodChannel*)methodChannel {
+  self = [super init];
+  if (self) {
+    [_methodChannel setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
+      if (!self) {
+        return;
+      }
+      [self handleMethodCall:call result:result];
+    }];
+    _textChecker = [[UITextChecker alloc] init];
+  }
+  return self;
+}
+
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+  NSString* method = call.method;
+  NSArray* args = call.arguments;
+  if ([method isEqualToString:kInitiateSpellCheck]) {
+    FML_DCHECK(args.count == 2);
```

Comment:
Assert both fields cannot be nil.
They can still be NSNull if dart passes null to this method channel. I added a bail below when either of them is NSNull